active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+```CS
+public event WindowActivatedEventHandler Activated
+```
+
+#### Samples
+
+```CS
+window.Actived += OnActivated;
+```	
+
+```CS
+void OnActivated(object sender, WindowActivatedEventArgs e)
+{
+   if (e.WindowActivationState == WindowActivationState.CodeActivated)
+    {
+        // 
+    }
+    else if (e.WindowActivationState == WindowActivationState.Deactivated)
+    {
+        // 
+    }
+    else if(e.WindowActivationState == WindowActivationState.PointerActivated)
+    {
+       //
+    }
+}
+```
+#### Remarks
+- **UWP**: Current behavior is [here](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.window.activated)
+
+### Window.Close method
+
+Closes and destroy the window. The Closed event occurs when a Window closes. 
+```CS
+public void Close()
+```
+
+#### Remarks
+- **UWP**: UWP app typically use a single Window instance, and do not open and close additional Window instances. Additionally, you will not typically provide custom UI to enable users to close the app window.
+
+### Window.Closed event
+
+Occurs after the window is closed. The Closed event is fired when the user closes the window, or the window is closed programmatically. Also, this causes the Activated event fires. 
+
+
+#### Samples
+
+```CS
+  window.Closed += OnClosed;
+```	
+```CS
+void OnClosed(object sender, MUX.WindowEventArgs e)
+{
+   //
+}
+```
+
+#### Remarks
+
+If this was the last window to be closed for the app:
+- **Win32**: The application ends and call Suspending event.
+- **UWP**: The application is suspended, and the Suspending event will fire. 
+
+### **Window placement management**
+
+### Window.SizeChanged event
+
+This event occurs whenever there are changes in the size of a window (e.g.: user resizes the window, user moves the window to a new display that has a different resolution and the window expands to fill it)
+
+```CS
+public event WindowSizeChangedEventHandler SizeChanged
+```
+#### Samples
+
+```CS  
+window.SizeChanged += OnSizeChanged;
+```	
+
+```CS
+void OnSizeChanged(object sender, MUX.WindowSizeChangedEventArgs e)
+{
+    if(!e.Size.IsEmpty && e.Size.Height != e.Size.Width)
+    {
+       //This is not a square Window
+    }
+}
+```
+#### Remarks
+- **UWP**: More info [here](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.window.sizechanged).
+
+### **Get the underneath window implementation**
+UWP apps use CoreWindow as a window implementation and Win32 apps use HWnds instead. The Microsoft.UI.Xaml.Window hides these detail implementations from the application developers and exposes a set of APIs that allow to create and manipulate Windows.
+These properties expose these underneath objects to enable advanced (or basic but not supported yet) scenarios, so developers can get the CoreWindow or HWnd objects to apply setters, listening events, etc.
+
+### Window.CoreWindow property: 
+
```

Comment:
Xaml's "Window" is going to become huge if it becomes a dumping ground for properties that will rarely be used. Can we move these to Hosting as a static member instead?

static class UwpWindowHosting
{
    static ApplicationView GetView(Window window);
    static CoreWindow GetCoreWindow(Window window);
}

Or, just have them call CoreWindow.GetForCurrentThread() and ApplicationView.GetForCurrentView() like everyone has done since Win 8.x.

Reply:
@jeffstall didn't you just say in the above comment
> There is no "current" window when we support multiple windows on a single thread

Is so, how would CoreWindow.GetForCurrentThread() work?

Reply:
I think his point is that rather than having instance members on the Window class, we should have static members on helper classes that take a Window instance as their first parameter.

We've done this for dumping grounds before (like [Typography](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Documents.Typography)), but in general it serves to make APIs undiscoverable. And I don't see Window becoming a dumping ground (WPF's [Window](https://docs.microsoft.com/dotnet/api/System.Windows.Window) is an existing precedent to look at).

Reply:
XAML's Window will live with Current for backwards compatibility. Perhaps we can deprecate it in the future, but not now.